src/sentry/seer/similarity/similar_issues.py

```diff
@@ -201,6 +203,23 @@ def get_similarity_data_from_seer(
             # which doesn't exist
             delete_seer_grouping_records_by_hash.delay(project_id, [parent_hash])
 
+            # Figure out how old the parent grouphash is, to determine how often this error is
+            # caused by a race condition.
+            parent_grouphash_age = None
+            parent_grouphash = (
+                GroupHash.objects.filter(project_id=project_id, hash=parent_hash)
+                .exclude(state=GroupHash.State.LOCKED_IN_MIGRATION)
+                .first()
+            )
```

Comment:
Use `.select_related` here to fetch the metadata row, so you avoid making two queries.

Another option is, since you're only using the `date_added`, you could just select that via `values_list` and avoid bringing the whole object over